graylog2-server/src/main/java/org/graylog2/shared/security/tls/DefaultTLSProtocolProvider.java

```diff
@@ -38,16 +38,20 @@
         }
 
         final Set<String> tlsProtocols = Sets.newHashSet(DEFAULT_TLS_PROTOCOLS);
+        final Set<String> supportedProtocols = getSupportedTlsProtocols();
+        if (tlsProtocols.retainAll(supportedProtocols)) {
+            LOG.warn("JRE doesn't support all default TLS protocols. Changing <{}> to <{}>", DEFAULT_TLS_PROTOCOLS, tlsProtocols);
+        }
+        defaultSupportedTlsProtocols = tlsProtocols;
+        return defaultSupportedTlsProtocols;
+    }
+
+    public synchronized static Set<String> getSupportedTlsProtocols() {
```

Comment:
What do you think about renaming the methods? I find it really hard to remember what the difference between `getSupportedTlsProtocols` and `getDefaultSupportedTlsProtocols` is. :smile: 

How about `getAllSupportedTlsProtocols` and `getSecureSupportedTlsProtocols`? (or something similar)

Reply:
good idea :+1: